### ISSUE TITLE
Support custom connection object types

### ISSUE THREAD

**Issue by bruce** [absinthe-graphql/absinthe_relay#11]
To support multiple connections to a node type, we need to support custom connection type names -- to support different properties for connection/edge types:

Something like, eg:

```elixir
connection :lived_locations, node_type: :location
connection :visited_locations, node_type: :location
```

```elixir
object :thing do
  # TODO: need some notation to use the right connection type...
  connection field :lived_locations, node_type: :location` do
    # ...
  end
  connection field :visited_locations, node_type: :location do
    # ...
  end
end
```

**Comment by austinsmorris** [comment]
Note that the edge type name should be customizable too.  It's the custom edges that have different properties.  But, the custom connection type has to specify the custom edge type.

**Comment by bruce** [comment]
@austinsmorris Custom edges should already be supported inside `edge do ... end`, eg, from the tests:

```elixir
     connection node_type: :pet do
        field :twice_edges_count, :integer do
          resolve fn
            _, %{source: conn} ->
              {:ok, length(conn.edges) * 2}
          end
        end
        edge do
          field :node_name_backwards, :string do
            resolve fn
              _, %{source: edge} ->
                {:ok, edge.node.name |> String.reverse}
            end
          end
        end
      end
```

The issue here is really addressability -- there's no way (currently) to define multiple connection types (and related edge types) for a node type, or to refer to those connection types when defining a `connection field`. Fixing that now :-)

**Comment by bruce** [comment]
Resolved by #13, will be released as part of v0.9.0.
